matrix/gauss-elim.js

```diff
@@ -80,8 +82,14 @@ function storeMatrix()
 {
     
     try {
-        matrices[0] = new Matrix($j("#matrix-entry").val().replace(/\n/g,";").replace(/;*$/,""));
-    } catch(err) { alert(err); return;}
+        // If the matrix has a \hline in it, turn on the appropriate button
+        if ((/\hline/).test($j("#matrix-entry").val())) {
+            settings.horizLine = true;
+            $j("#horizLine").attr("checked", settings.horizLine);
+        }
+        // Same filter as before, but additionally handles \\, \hline, and & surrounded by any number of spaces. 
+        matrices[0] = new Matrix($j("#matrix-entry").val().replace(/\s*&\s*/g, " ").replace(/\\\\/g, "").replace(/\n/g, ";").replace(/;*$/, "").replace(/\\hline/g, ""));

```

Comment:
This should be put into the matrix-operations.js file.  Probably right at the top.